Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3125,45 +3142,12 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             actionSheetController.addAction(openInNewTabAction, accessibilityIdentifier: "linkContextMenu.openImageInNewTab")
 
             let saveImageAction = UIAlertAction(title: Strings.saveImageActionTitle, style: .default) { _ in
-                self.getData(url) { [weak self] data in
-                    guard let self = self, let image = UIImage(data: data, scale: UIScreen.main.scale) else {
-                        return
-                    }
-                    
-                    UIImageWriteToSavedPhotosAlbum(image, self, #selector(self.image(image:didFinishSavingwithError:contextInfo:)), nil)
-                }
+                self.writePhotoToAlbumAction(url)
```

Comment:
make `self` weak, maybe it doesnt make much sense in context of browserViewController :)